### ISSUE TITLE
⌨ Table "Fitler by" dropdown does not work with NVDA

### ISSUE THREAD

**Issue by esizer** [GCTC-NTGC/gc-digital-talent#12454]
## ⌨ Accessibility Issue

When using NVDA, the dropdown to select which column you would like to search by does not function....at all.

## 🦋 Expected Behaviour

Users should be able to open the dropdown with either <kbd>Enter</kbd> or <kbd>Space</kbd> and then cycle through the buttons using array keys.

## 🕵️ Details

h/t to @vd1992 for discovering this!

This has me stumped. It works in every other screen reader and I can't see anything wrong with the HTML, it even announces it as a dropdown that can be collapsed/opened 😖 Also, other dropdowns still function and they use the same underlying component :/

- **Operating System**: Windows 11
- **Browser**: Firefox + Chrome
- **Assistive Technology**: NVDA

## 📋 Steps to Reproduce

1. Login as admin `admin@test.com`
2. Navigate to any table with a search bar `/admin/users` is a good one!
3. Attempt to select a specific column under "Filter by"
4. Observe that it is impossible

## 🙋‍♀️ Proposed Solution

None 😭 

## ✅ Acceptance Criteria

- [ ] You can use the filter by dropdown with NVDA active

**Comment by vd1992** [comment]
Note, can't be reached via keyboard navigation
